### ISSUE TITLE
Feature Request/ 4-Point perspective transformation 

### ISSUE THREAD

**Issue by wpaholic** [fabricjs/fabric.js#6769]
Here's vanilla js version i've just found on stackoverflow https://stackoverflow.com/a/30088700

It would be great if this feature can be implemented officially into fabric for image object (or any object) in future versions. i know someone has already created 8-point here https://malipetek.github.io/grocery/fabric-warp but it's outdated.

Could you guys please consider this feature? i bet everyone else is also still waiting for this.

Wish the authors, contributors and everyone Happy Holidays.

**Comment by asturur** [comment]
I have a fabric imementation of that that i m not using anymore at work. It works
like a graphic filter.
To be honest isn't that much useful and prone fo become a mess in few moves. I ll ask if i can release it.

**Comment by wpaholic** [comment]
The above 4point transform example from stackoverflow look promising, and i think something like css transform skewZ is good enough, i assume we have ABCD rectangle and when moving point A to the BD-axis, we can restrict it at the axis and we don't have to calculate the curve and stuff.

when combining with clipPath, this feature would be really useful to create composite objects to make 'em look more depth.

**Comment by asturur** [comment]
This feature will have an hard time working well with the other, being a filter and not a proper geometry of the 2d canvas.
If i can provide my implementation i will, but it won't be a part of the standard library.
I already know i will have an hard time trying to have it make sense.

**Comment by pt-tal** [comment]
I really want to achieve this feature.

**Comment by asturur** [comment]
i have a bunch of code that enables this as an image filter

**Comment by taherPardawala** [comment]
Hey @asturur, did you end up actually sharing your code? I have a similar use case and would like to see how you have implemented it.

**Comment by noctivityinc** [comment]
@asturur I second that.  Please share any perspective code you got working.  It we could fill a polygon with an image this seems like a no-brainer to make work, but not sure we can do that right now.

**Comment by 3zzy** [comment]
@asturur Please share the code if you can. Thanks

**Comment by asturur** [comment]
Yes i can share now, is ugly, but i think can be at least an inspiration

**Comment by thanhdinh98** [comment]
Hi @asturur 
Please share the code, I'm looking for solution for this. Thank you for your great work!

**Comment by asturur** [comment]
I m osorry is very hard for me to find the time

**Comment by thanhdinh98** [comment]
No worry @asturur, I understand, keep up the good work mate.
btw, I would like to share a easy alternative solution for this problem. I've created a custom Image object works with perspectivejs. The function seems work at the moment. I tested on fabric version 4.3.0
Here's the [code]( https://gist.github.com/thanhdinh98/b89cda641de0d01cfcaf0783272407d4)
This code can be updated adding the points to make the image draggable.
Here's the demo
![image](https://github.com/fabricjs/fabric.js/assets/25563537/a8ff63fd-3604-4528-9597-f0ae2e8fb857)
